Master

brian0305 opened the pull request.

brian0305 committed:
Damage :dancer:

_ Fix bug OnLoad and improve Bootstrap (make sure Bootstrap finish b4 do
other thing)
_ Fix AA name of Gangplank
_ Update all Damage
_ Fix name of champion in TargetSelector, add check Kindred R, fix get
valid target

brian0305 committed:
Revert something :(

brian0305 committed:
Add IE to all champ

brian0305 committed:
Revert

brian0305 committed:
Add some stuff

brian0305 committed:
Merge remote-tracking branch 'refs/remotes/LeagueSharp/master'

brian0305 committed:
Yeh

brian0305 committed:
Yeh

brian0305 committed:
Remove Lee R Collided dmg

Screeder commented on Core/Orbwalker.cs:
```diff
@@ -251,6 +263,16 @@ public static AttackableUnit GetTarget(OrbwalkerMode? modeArg)
+                        .Where(
+                            m =>
+                            m != null && m.CharData.BaseSkinName == "gangplankbarrel" && m.IsHPBarRendered
+                            && m.Health < 2))
```
not health == 1?

brian0305 replied:
< 2 what diff with == 1 :(

Screeder replied:
so it won't check for 0 because it should never be null. I only think that it would be better to be precisely.

brian0305 replied:
is Barrel have 0 health :(

Screeder commented on Core/Utils/AutoAttack.cs:
```diff
@@ -131,6 +132,12 @@ public static float GetRealAutoAttackRange(this AttackableUnit target)
             {
+                var targetBase = target as Obj_AI_Base;
+                if (targetBase != null && GameObjects.Player.ChampionName == "Caitlyn"
+                    && targetBase.HasBuff("caitlynyordletrapinternal"))
```
also applies for cait e?

brian0305 replied:
I think it does bcz i steal it from Sebby and Xcsoft, both of them only use that buff name

Screeder commented on Core/Wrappers/Damages/Damage.cs:
```diff
@@ -386,6 +506,15 @@ private static double CalculatePhysicalDamage(this Obj_AI_Base source, Obj_AI_Ba
                         armorPenetrationPercent = 0.3d;
                         break;
                 }
+                if (target is Obj_AI_Minion)
```
case TurretType.TierTwo:
                         armorPenetrationPercent = 0.7d;

shouldn't it be 0,75?

Screeder commented on Core/Wrappers/Damages/Damage.cs:
```diff
@@ -386,6 +506,15 @@ private static double CalculatePhysicalDamage(this Obj_AI_Base source, Obj_AI_Ba
                 }
+                if (target is Obj_AI_Minion)
+                {
+                    amount *= 1.25;
```
stacks up to 1.25 after some time

brian0305 replied:
https://github.com/LeagueSharp/LeagueSharp.Common/commit/b1fafbfaa36b4df2e4f37ba32b6a731a879d271e
I base on this, if sth wrong then ask Honda ^^

brian0305 committed:
R# + Fix dmg calc for Turret

brian0305 committed:
Add Urgot R (Dmg Reduce)

Screeder commented on Core/Wrappers/Damages/Damage.cs:
```diff
@@ -473,18 +574,34 @@ private static double CalculatePhysicalDamage(this Obj_AI_Base source, Obj_AI_Ba
+                // + IN THIS TOGETHER 8 % of the damage that the nearest allied champion would take is dealt to you instead.This can't bring you below 15% health.
+                if (targetHero.GetResolve(DamageMastery.Resolve.BondofStone).IsValid())
+                {
+                    amount *=
```
not 1000 range?
http://leagueoflegends.wikia.com/wiki/Bond_of_Stone_(Season_2016_Mastery)

Screeder commented on Core/Wrappers/Damages/Damage.cs:
```diff
@@ -493,68 +610,68 @@ private static double CalculatePhysicalDamage(this Obj_AI_Base source, Obj_AI_Ba
+                    amount -= new[] { 2, 4, 6, 8, 10 }[targetHero.Spellbook.GetSpell(SpellSlot.E).Level - 1];
                 }
 
                 // Unbreakable
```
mhh hwo to implement first hit is 100% dmg reduction? buff changes or so?

Screeder commented on Core/Wrappers/Damages/DamagePassives.cs:
```diff
@@ -357,69 +474,71 @@ private static void CreatePassives()
+                                    var spellW =
+                                        ((Obj_AI_Hero)@base.GetBuff("kalistacoopstrikemarkbuff").Caster).Spellbook
+                                            .GetSpell(SpellSlot.W).Level - 1;
+                                    return Math.Min(
```
check for minion first line or champion second line dmg

brian0305 committed:
Fix Kalis W + Mastery Range + Commented Ori Passive

brian0305 committed:
Make GetMastery to be public

brian0305 committed:
zxc

Orbwalker:
+ Remove useless stuff
+ Add new Jungle mob
Item:
+ Fix ItemID for GetWardSlot
Damage:
+ Optimize code
+ Add reduce dmg when atk new jungle mob

brian0305 committed:
Track into Crit + Fix GetAADmg logic

brian0305 committed:
zcv

brian0305 committed:
zxcáq

brian0305 committed:
Finish

brian0305 committed:
fix regex
